### PR TITLE
Correct quick fix message for `W605`

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR fixes the `W605` rule implementation to provide the quickfix message as
per the fix provided.

## Test Plan

Update snapshots.

fixes: #8155

### CODE REVIEW

**Comment by dhruvmanila** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8154** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8154" target="_blank">Graphite</a> 
    * **PR #8156** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8156" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/8156?utm_source=stack-comment).

**Comment by dhruvmanila** [comment]
I was working on `W605` for the f-string change so just thought to stack it up on top of that.

File: crates/ruff_linter/src/rules/pycodestyle/rules/invalid_escape_sequence.rs [hunk 1]
```diff
@@ -28,20 +28,38 @@ use crate::fix::edits::pad_start;
...
-        format!("Invalid escape sequence: `\\{char}`")
+        let InvalidEscapeSequence { ch, .. } = self;
+        format!("Invalid escape sequence: `\\{ch}`")
     }
 
     fn fix_title(&self) -> String {
-        "Add backslash to escape sequence".to_string()
+        match self.fix_title {
+            FixTitle::AddBackslash => format!("Add backslash to escape sequence"),
+            FixTitle::TurnIntoRawString => format!("Turn into raw string"),
```
Comment (zanieb) [inline crates/ruff_linter/src/rules/pycodestyle/rules/invalid_escape_sequence.rs:46]: I'd prefer `UseRawStringLiteral` and "Use a raw string literal" instead

File: crates/ruff_linter/src/rules/pycodestyle/rules/invalid_escape_sequence.rs [hunk 2]
```diff
@@ -28,20 +28,38 @@ use crate::fix::edits::pad_start;
...
-        let InvalidEscapeSequence(char) = self;
-        format!("Invalid escape sequence: `\\{char}`")
+        let InvalidEscapeSequence { ch, .. } = self;
+        format!("Invalid escape sequence: `\\{ch}`")
     }
 
     fn fix_title(&self) -> String {
-        "Add backslash to escape sequence".to_string()
+        match self.fix_title {
+            FixTitle::AddBackslash => format!("Add backslash to escape sequence"),
```
Comment (zanieb) [inline crates/ruff_linter/src/rules/pycodestyle/rules/invalid_escape_sequence.rs:45]: Should we add one of this fix to the examples?

Comment (dhruvmanila) [reply crates/ruff_linter/src/rules/pycodestyle/rules/invalid_escape_sequence.rs:55]: Yeah, I've added it. Thanks

**Review by zanieb — APPROVED** [review]

**Comment by zanieb** [comment]
@dhruvmanila sorry I was very naughty and broke your stack thinking this went into `main` 😬

**Comment by dhruvmanila** [comment]
Oh lol, no worries.

**Pull request merged**